sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/HttpClient.java

```diff
@@ -20,11 +21,31 @@ public interface HttpClient {
     Response<?> send(HttpRequest request);
 
     /**
-     * Creates a new {@link HttpClient} instance.
+     * Get a new instance of the {@link HttpClient} that the default {@link HttpClientProvider} is configured to
+     * create.
      *
-     * @return A new {@link HttpClient} instance.
+     * <p>If no {@link HttpClientProvider} can be found on the classpath, a new instance of the default
```

Comment:
```suggestion
     * <p>If no {@link HttpClientProvider} can be found on the classpath, a new instance of the {@link DefaultHttpClientProvider default}
```

Reply:
We shouldn't reference implementation classes on our public Javadoc.